modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -219,6 +219,22 @@ Error GDScriptAnalyzer::check_class_member_name_conflict(const GDScriptParser::C
 	return OK;
 }
 
+void GDScriptAnalyzer::get_script_classes(GDScriptParser::ClassNode *p_node, RBSet<GDScriptParser::ClassNode *> &p_set) {
+	if (p_set.has(p_node)) {
+		return;
+	}
+	p_set.insert(p_node);
+
+	// Prioritize node base type over its outer class
```

Comment:
I'm not 100% on red-black trees, but don't they reorder stuff around? If that's the case, then this wouldn't actually prioritize by insertion order.

Reply:
Maybe. I'll use a vector instead.

Reply:
Are there ordered hash sets in Godot? I think that'd be the ideal so the check isn't O(n), but honestly, this vector is never going to be very big so I think efficiency is not paramount :)